python/traverse.py

```diff
@@ -0,0 +1,63 @@
+from typing import Callable, Union
+
+import lxml.html
+
+
+def traverse_and_execute(
+    html: str, fn: Callable[[lxml.html.HtmlElement], None]
```

Comment:
If we're doing more intricate things with lxml elements now, we might want to include type stubs of some type (either externally or our own). Right now everything from lxml is treated as  Any because the stubs are missing.

Reply:
I ran headfirst into two issues with the stubs:

- https://github.com/lxml/lxml-stubs/issues/81
- https://github.com/lxml/lxml-stubs/issues/68

Given that, I'll opt to omit the stubs and forge ahead with `Any`s 😢

Reply:
Yeah, I ran into similar problems when I tried both of the public stubs packages. We'll probably have to figure out how to add our own if we want to use them. We only use a small portion of the library so it shouldn't be a huge deal, but I ran into issues getting the directory right.

Reply:
Sorry to chip in the discussion without any pretext. I have been maintaining [an enhanced lxml stub](https://github.com/abelcheung/types-lxml), does it solve the first issue?

For 2nd issue, actually `_Element.getchildren()` has been deprecated for a long time, that's why it doesn't appear in any stub. Current way is to use `list(_Element)` to get all children. Hope this helps!

Reply:
I noticed this was linked from [lxml-stubs](https://github.com/lxml/lxml-stubs/issues/68) and recognized some folks from UIUC :wave:

That particular issue is about lack of stubs for `getchildren()`, but `getchildren()` is deprecated anyway (TIL).

After removing the `getchildren()`s, the rest of the `lxml-stubs` were mostly fine/usable with Mypy for my project, but also I'm not parsing HTML, so grains of salt.